lib/internal/streams/pipeline.js

```diff
@@ -102,25 +106,50 @@ async function* fromReadable(val) {
 }
 
 async function pump(iterable, writable, finish) {
-  if (!EE) {
-    EE = require('events');
-  }
   let error;
+  let callback = noop;
+  const resume = (err) => {
+    error ||= err;
```

Comment:
If it is possible that several errors are sent to the stream, we could aggregate them instead of swallowing subsequent errors. Please disregard if they are swallowed on purpose.

```suggestion
    error = aggregateTwoErrors(error, err);
```

Reply:
Note that you need to import `aggregateTwoErrors` from `internal/errors`.